packages/core/src/types/context.ts

```diff
@@ -140,13 +141,17 @@ export type KeystoneDbAPI<KeystoneListsTypeInfo extends Record<string, BaseListT
 
 export type KeystoneGraphQLAPI = {
   schema: GraphQLSchema;
-  run: (args: GraphQLExecutionArguments) => Promise<Record<string, any>>;
-  raw: (args: GraphQLExecutionArguments) => Promise<ExecutionResult<Record<string, any>>>;
+  run: <TData = any, TVariables = Record<string, any>>(
+    args: GraphQLExecutionArguments<TData, TVariables>
+  ) => Promise<ExecutionResult<TData, TVariables>['data']>;
+  raw: <TData = any, TVariables = Record<string, any>>(
+    args: GraphQLExecutionArguments<TData, TVariables>
+  ) => Promise<ExecutionResult<TData, TVariables>>;
```

Comment:
```suggestion
  run: <TData, TVariables = Record<string, any>>(
    args: GraphQLExecutionArguments<TData, TVariables>
  ) => Promise<ExecutionResult<TData, TVariables>['data']>;
  raw: <TData, TVariables = Record<string, any>>(
    args: GraphQLExecutionArguments<TData, TVariables>
  ) => Promise<ExecutionResult<TData, TVariables>>;
```

We should require people to cast the result if they're not using a typed document node